test/WebJobs.Script.Tests/Security/SecretManagerTests.cs

```diff
@@ -54,6 +55,88 @@ public SecretManagerTests(ITestOutputHelper outputHelper)
             _startupContextProvider = new StartupContextProvider(_testEnvironment, loggerFactory.CreateLogger<StartupContextProvider>());
         }
 
+        [Fact]
+        public void SecretManager_GenerateIdentifiableSecret_HonorsSeed()
+        {
+            ulong seed = uint.MaxValue;
+            string secret = SecretManager.GenerateIdentifiableSecret(uint.MaxValue);
+            ValidateSecret(secret, seed);
+        }
+
+        [Fact]
+        public async Task SecretManager_NewlyGeneratedKeysAreIdentifiable()
+        {
+            using (var directory = new TempDirectory())
+            {
+                string startupContextPath = Path.Combine(directory.Path, Guid.NewGuid().ToString());
+                _testEnvironment.SetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteStartupContextCache, startupContextPath);
+                _testEnvironment.SetEnvironmentVariable(EnvironmentSettingNames.WebSiteAuthEncryptionKey, TestEncryptionKey);
+
+                // Because we don't initialize any disk context, we can configure
+                // the secret manager to allocate fresh keys on start-up.
+                using (var secretManager = CreateSecretManager(directory.Path, createHostSecretsIfMissing: true))
+                {
+                    var hostSecrets = await secretManager.GetHostSecretsAsync();
+
+                    // Here and elsewhere we elide the '!' character which is prepended to every
+                    // generated secret value, as a clue that actual encryption is simulated in
+                    // the test case.
+                    ValidateSecret(hostSecrets.MasterKey.Substring(1), SecretManager.MasterKeySeed);
+
+                    // Create host secrets if missing knob does not allocate a system
+                    // key. The system key creation/validation test is done in the
+                    // DefaultScriptWebHookProvider tests.
+                    Assert.True(hostSecrets.SystemKeys.Count == 0);
+
+                    foreach (string key in hostSecrets.FunctionKeys.Values)
+                    {
+                        ValidateSecret(key.Substring(1), SecretManager.FunctionKeySeed);
+                    }
+                }
+            }
+        }
+
+        private static void ValidateSecret(string secret, ulong seed)
+        {
+            // All Azure function keys are 40 bytes in length, 56 chars
+            // when base64-encoded. A URL friendly encoding elides the
+            // '==' padding, however.
```

Comment:
I don't understand this comment - SecretManager.GenerateIdentifiableSecret is appending == when not present.